core/jobargument.go

```diff
@@ -75,56 +66,61 @@ func (a *JobArgument) Append(s string, isS3path bool) *JobArgument {
 	return a
 }
 
-func (a *JobArgument) CheckConditionals(wp *WorkerParams, src *JobArgument, opts opt.OptionList) (ret error) {
+// CheckConditions checks if the job satisfies the conditions if the job has -n, -s and -u flags.
+// It returns error-embedded JobResponse with status "warning" if none of the requirements are met.
+// It returns nil if any warning or error is encountered during this check.
+func (a *JobArgument) CheckConditions(wp *WorkerParams, src *JobArgument, opts opt.OptionList) *JobResponse {
```

Comment:
Since you're already here, could you separate this method from JobArgument and make it a function please? A function like 

`CheckConditions(src, dst *JobArgument, wp, opts) *JobResponse` makes it clear that `a` is the destination here.

Reply:
Done